go/vt/topo/topoproto/tablet.go

```diff
@@ -75,19 +76,23 @@ func TabletAliasUIDStr(ta *topodatapb.TabletAlias) string {
 	return fmt.Sprintf("%010d", ta.Uid)
 }
 
+const tabletAliasFormat string = "([-_.a-zA-Z0-9]+)-([0-9]+)"
```

Comment:
This could be made a little nicer with named capture groups and `SubexpIndex`:

```suggestion
const tabletAliasFormat string = "(?P<cell>[-_.a-zA-Z0-9]+)-(?P<uid>[0-9]+)"
```

Reply:
In addition, is there a reason you don't add the anchors here instead of adding them inline when you create `tabletAliasRegexp`?

Reply:
The original intention was to print out the `tabletAliasFormat` in the error message, in which case dropping the anchors is more friendly to human eyes. I do realize I never executed my plan, hence the confusion 😄 


If everyone thinks this is a welcome change, I'm happy to commit it.

* old error message: `invalid tablet alias: '%s', expecting format: '<cell>-<uid>'`
* new error message `invalid tablet alias: '%s', expecting format: '(?P<cell>[-_.a-zA-Z0-9]+)-(?P<uid>[0-9]+)'`

Reply:
Ah that thinking makes sense. My one concern is that if we're matching on an anchored regex, and we produce an error that suggests we're not that that could lead to confusion. For an (admittedly trivial) example `invalid tablet alias: ',zone1-100', expecting format '(?P<cell>[-_.a-zA-Z0-9+)-(?P<uid>[0-9]+)'` would still require me to read the code to confirm the regexp is anchored (although in this example it wouldn't be too hard to guess that), whereas if we include the anchors then that question is already answered for the user.

Either way, including the regexp in any form in the error string seems like a great idea to me!

Reply:
Change committed! ✅